### ISSUE TITLE
Travis build is taking too long

### ISSUE THREAD

**Issue by hvesalai** [slick/slick#1977]
Currently all the tests for all the different Scala versions are run in one Travis job. Travis has a timeout of 50 minutes for each job, which we are now hitting.

However, Travis does not have any limit for the build time as long as the build consists of one or more jobs that each are less than 50 minutes.

What we need to do is to split the current build into a [Build Matrix](https://docs.travis-ci.com/user/build-matrix/) of Scala versions × DBs, i.e. have each DB test for each scala version be a different job in the build.

**Comment by hvesalai** [comment]
@trevorsibanda @renatocaval or anybody who knows anything about Travis, we need this before we can release 3.3 since currently all our builds are failing.

**Comment by renatocaval** [comment]
Ok, I can take a look. Not Travis expert here, but Lagom and Play builds are split in small jobs. I can get some inspiration from there.

**Comment by smootoo** [comment]
Just looking at this, I think one of my PRs might have pushed us close to the limit. #1960 
Let me check and I'll reduce the number of tests that are performed.

**Comment by renatocaval** [comment]
Yep, that can be the case. Eventually, we can temporarily reduce the number of iterations there. But we should definitely improve the build.

I will continue work on this one, but eventually, we can unlock 3.3.

**Comment by smootoo** [comment]
Still waiting to see how long the travis builds are on my fork, but I've raised the PR #1978 anyway.
Let's see if it gets us back inside the limit.

**Comment by hvesalai** [comment]
Ok, let's postpone this.

**Comment by WellingR** [comment]
@hvesalai now that travis builds the scala versions x jdk versions separately. So we still need to run separate builds per database? Right now it takes about 20-25 minutes per build.

**Comment by hvesalai** [comment]
I guess we can live with the current thing for now and close this. If the problem reappears, we reopen this or a new issue.
